### PR TITLE
APEXCORE-674 Change access specifier of DTConfiguration.ValueEntry to private

### PR DESCRIPTION
**Pull request by ajaygit158-zz**

### CODE REVIEW

**Comment by ajaygit158-zz** [comment]
@vrozov Kindly review

**Review by vrozov — APPROVED** [review]
Looks good to me. The only concern I have is that `set` silently overrides scope with the Scope.LOCAL. As this is an existing behavior, it can be handled in a separate JIRA/PR.

File: engine/src/main/java/com/datatorrent/stram/client/DTConfiguration.java [hunk 1]
```diff
@@ -275,36 +275,32 @@ public void remove(String key)
...
+  public void setInternal(String key, String value)
   {
-    ValueEntry valueEntry;
-    if (map.containsKey(key)) {
-      valueEntry = map.get(key);
+    ValueEntry valueEntry = map.get(key);
+    if (valueEntry != null) {
       valueEntry.value = value;
     } else {
       valueEntry = new ValueEntry();
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/client/DTConfiguration.java:284]: @ajaygit158 Not directly related to the PR, but once you are on it, please either remove setInternal() or fix it to update `value` in the case when `key` is not present in the map. In the later case, please add unit test.

Comment (ajaygit158-zz) [reply engine/src/main/java/com/datatorrent/stram/client/DTConfiguration.java:284]: @vrozov I assume you meant when key is absent (else clause). Added the unit test.

Comment (vrozov) [reply engine/src/main/java/com/datatorrent/stram/client/DTConfiguration.java:284]: Right.

**Pull request merged**